operator/v1/types_ingress.go

```diff
@@ -1684,3 +1699,50 @@ type IngressControllerList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []IngressController `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+//
+// Ingress contains configuration options specific to the Ingress Operator itself.
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
```

Comment:
I added `+openshift:compatibility-gen:level` because it seems to be expected in the root CRDs, but it's not something I think I understand yet, beyond what is said in the comment. Is there a process I should be following to determine the appropriate compatibility level, or a rule of thumb I can use?

Reply:
For a v1 API, level 1 is required and changing it to any other level will cause a validation error. So, not much more to do here right now